#! /bin/sh -e
## changed_hiscore_name.dpatch by Anibal Avelar <aavelar@cofradia.org>
##
## DP: apply patch to change the hiscore name

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argum
ent"
                exit 1;;
esac

exit 0
diff -bBdNrw -U5 njam-1.25/src/njam.cpp njam-1.25.modif/src/njam.cpp
--- njam-1.25/src/njam.cpp	2005-12-13 05:56:43.000000000 -0600
+++ njam-1.25.modif/src/njam.cpp	2008-05-12 11:59:46.694847801 -0500
@@ -235,11 +235,11 @@
 		fprintf(fp, "I=%s\n", m_GameOptions.ServerIP);
 		fclose(fp);
 	}
 
 		// format: NAME#POINTS#LEVEL#
-	fp = fopen("hiscore.dat", "w+");
+	fp = fopen("njam-hiscore.dat", "w+");
 	if (fp)
 	{
 		for (int i=0; i<10; i++)
 			fprintf(fp, "%s#%d#%d#\n", TopTenScores[i].name, TopTenScores[i].points, TopTenScores[i].level);
 		fclose(fp);
@@ -647,11 +647,11 @@
 	}
     else
         LogFile("Failed to open conf file.\n");
 
 	// create default hiscore
-    LogFile("Creating default hiscore.\n");
+    LogFile("Creating default njam-hiscore.\n");
 	char DefaultNames[10][10] = {
 		"MILAN", 		"TANJA",
 		"DULIO", 		"DJORDJE",
 		"CLAUS",		"ENZO",
 		"JOLAN",		"JAAP",
@@ -663,14 +663,14 @@
 		sprintf(TopTenScores[i].name, "%s\0", DefaultNames[i]);
 	}
 
 	// load hiscore from file (if any)
 	// format: NAME#POINTS#LEVEL#
-	fp = fopen("hiscore.dat", "r");
+	fp = fopen("njam-hiscore.dat", "r");
 	if (fp)
 	{
-		LogFile("Reading hiscore.dat\n");
+		LogFile("Reading njam-hiscore.dat\n");
 		char buff[40];
 		int number = 0;
 		while (!feof(fp) && number < 10)
 		{
 			fgets(buff, 40, fp);
